22-discovery.ks

```diff
@@ -1,5 +1,8 @@
 %post
 
+gem install newt --no-document
+gem install fast_gettext --no-document
```

Comment:
I'd strongly object to using `gem install`. It makes builds (using the same buildroot) non-reproducable. I also think in various environments you don't even have network access available

Reply:
Well it's the way how to get rid of the meta-package in the foreman-packaging

> I also think in various environments you don't even have network access available

It's usually us who is building the image, so I don't see it as such a problem.

Reply:
Plus there are multiple steps in the process where the internet connection is required, without it it's impossible to build the FDI.

Reply:
I'll hard block on it. If you want to get rid of it, you must list `rubygem-newt` and `rubygem-fast_gettext` in the packages section.

Reply:
Another concern that is perhaps more related to downstream: they calculate the set of packages to include in a compose. I'd check with them if it doesn't break anything when you remove the meta-package.

> It's usually us who is building the image, so I don't see it as such a problem.

https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#Building_a_custom_Discovery_Image_provisioning

Downstream this chapter used to exist, but it was hidden because it's broken. That doesn't mean we shouldn't support it.

Reply:
> I'll hard block on it. If you want to get rid of it, you must list rubygem-newt and rubygem-fast_gettext in the packages section.

Will take a look into that.